packages/jest-core/src/__tests__/watch-file-changes.test.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import path from 'path';
+import fs from 'fs';
+import {JestHook} from 'jest-watcher';
+import Runtime from 'jest-runtime';
+import {normalize} from 'jest-config';
+import HasteMap from 'jest-haste-map';
+import {AggregatedResult} from '@jest/test-result';
+
+describe('Watch mode flows with changed files', () => {
+  let watch: any;
+  let pipe: NodeJS.ReadStream;
+  let stdin: MockStdin;
+  const fileTargetPath = path.resolve(
+    __dirname,
+    '__fixtures__',
+    'lost-file.js',
+  );
+  const fileTargetPath2 = path.resolve(
+    __dirname,
+    '__fixtures__',
+    'watch-test.test.js',
+  );
+  const cacheDirectory = path.resolve(__dirname, `tmp${Math.random()}`);
+  let hasteMapInstance: HasteMap;
+  const deleteFolderRecursive = pathname => {
+    if (fs.existsSync(pathname)) {
+      fs.readdirSync(pathname).forEach(file => {
+        const curPath = path.resolve(pathname, file);
+        if (fs.lstatSync(curPath).isDirectory()) {
+          // recurse
+          deleteFolderRecursive(curPath);
+        } else {
+          // delete file
+          fs.unlinkSync(curPath);
+        }
+      });
+      fs.rmdirSync(pathname);
+    }
+  };
+
+  beforeEach(() => {
+    watch = require('../watch').default;
+    pipe = {write: jest.fn()} as any;
+    stdin = new MockStdin();
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+    hasteMapInstance.end();
+    [fileTargetPath2, fileTargetPath].forEach(file => {
+      try {
+        fs.unlinkSync(file);
+      } catch (e) {}
+    });
+    deleteFolderRecursive(cacheDirectory);
+  });
+
+  it('should correct require new files without legacy cache', async () => {
+    fs.writeFileSync(
+      fileTargetPath2,
+      `
+        require('${fileTargetPath}');
+        describe('Fake test', () => {
+            it('Hey', () => {  
+            
+            });
+        });
+      `,
+      {
+        encoding: 'utf-8',
+      },
+    );
+
+    fs.mkdirSync(cacheDirectory);
+    const config = normalize(
```

Comment:
`TestUtils` can probably help make this a bit smaller

Reply:
I removed needless properties